discovery/src/main/scala/akka/discovery/SimpleServiceDiscovery.scala

```diff
@@ -54,6 +55,29 @@ object SimpleServiceDiscovery {
         case None => host
       }
   }
+
+  /**
+   * Represents a service discovery lookup.
+   *
+   * Each mechanism should document what a Simple vs Full lookup means.
+   * Implementations are free to ignore port/protocol from a Full lookup
+   * but should not throw if they are set unexpectedly as this prevents the
+   * same query being sent to multiple mechanisms via the aggregate service
+   * discovery
+   */
+  @ApiMayChange
+  sealed abstract class Lookup {
+    def name: String
+  }
+
```

Comment:
I now realised that these Simple/Full may be better perhaps as factory methods on `Lookup` -- instead of the separate classes?